src/search-backend/search-backend-factory.ts

```diff
@@ -0,0 +1,40 @@
+import { FulltextSearchBackend } from './fulltext-search-backend';
+import { MetadataSearchBackend } from './metadata-search-backend';
+import { SearchType } from '../search-type';
+import type { SearchBackendInterface } from './search-backend-interface';
+import type { SearchBackendOptions } from './search-backend-options';
+import { Memoize } from 'typescript-memoize';
+
+/**
+ * A factory to obtain the correct search backend for a given search type.
+ */
+export class SearchBackendFactory {
+  private constructor() {
+    //
+  }
+
+  /**
+   * Retrieve a search backend that can handle the given type of search.
+   * @param type The type of search that the backend needs to handle.
+   * @param options Options to pass to the search backend.
+   */
+  @Memoize((type: SearchType, options: SearchBackendOptions = {}) => {
+    // We can memoize backends based on their params, to avoid constructing redundant backends
+    const { includeCredentials = '', scope = '', baseUrl = '' } = options;
+    return `${type};${includeCredentials};${scope};${baseUrl}`;
+  })
+  static getBackendForSearchType(
```

Comment:
Q: what do you think rolling this method up to `search-service`?

Reply:
Yeah, that would probably be cleaner overall. Thanks for the suggestion!